pallets/loans/src/lib.rs

```diff
@@ -526,7 +526,7 @@ pub mod pallet {
 		/// Loan is accrued before transferring the amount to reserve.
 		/// If the repaying amount is more than current debt, only current debt is transferred.
 		/// Amount of token will be transferred from owner to Pool reserve.
-		#[pallet::weight(<T as Config>::WeightInfo::repay(T::MaxActiveLoansPerPool::get()))]
+		#[pallet::weight(<T as Config>::WeightInfo::repay(T::MaxActiveLoansPerPool::get(), MaxRateCountOf::<T>::get()))]
```

Comment:
It feels like it's a lot of knowledge for the loans pallet to handle the `MaxRateCount` coming from a trait in a method that does not make use explicitly anything related to this count. The only way to know you must put this weight here is by looking deeper into how `InterestAccrual` implementation you've configured to see how it works.

In fact, the guilty method that forces you to add this is `interest_accrual::Pallet::get_rate()`, which is used by the implementation of `InterestAccrual::current_debt()` used in `price()`. Seems like a very long path to realizing about this.

I understand that it's needed for the benchmarks, but not sure if the complexity it adds here deserves to be measured it nowdays. Probably the Substrate framework should offer some kind of utility for this kind of things 🥲

Reply:
I agree it's a lot of knowledge. If we come up with a better way to abstract this I would love to implement it